### DISCUSSION TITLE
Tesla Browser audio too low

### DISCUSSION THREAD

**Discussion by ShodanOne** [tesla-android/issue-tracker#74] [Q&A]
Hello, I’m using 2022.27.1.
It works fine, my main comment is that tesla browser audio is too low, so sounds from app through carplay can hardly be heard when music is playing through bluetooth or Tesla OS, since tesla android cannot lower the volume with app notifications.
Is there a way to setup the audio volume higher (my tesla android volume us already at maximum), or maybe do something at the audio capture app level ?
Thank you.

**Comment by Cooker44** [comment]
Edit: OP is correct

I’m afraid I don’t have an answer for browser volume, but I can say that shouldn’t CarPlay sounds be coming via Bluetooth? Have you set Carlinkit audio to Bluetooth in settings? Phone then connects to car Bt and all CarPlay audio comes through with excellent volume (better than Spotify app in the Tesla os actually)

**Reply by ShodanOne** [reply]
Hello,
My concern is more for navigation sounds (Waze, Coyote), I have no issue with music which seems to go directly through the phone-car Bluetooth link, disregarding how the « Audio Channel » option is set in the carlink options.
According to the installation guide « Navigation sounds from CarPlay will be routed via Tesla Browser even when Audio Channel is set to Bluetooth in Autokit. ».

**Reply by Cooker44** [reply]
You are dead right….sorry. Used waze yesterday and tested…volume extremely low and not audible if music playing.

**Reply by galosre** [reply]
Just the opposite for me, Wase sound extremely loud

**Reply by yerode** [reply]
I habe the same problem. Audio volume from the browser and Tesla Android - compared to e.g. DVB radio - is comparably low. You cannot hear any guidance commands from Google or ABRP when listening to Tesla DVB radio.

**Reply by mikegapinski** [reply]
@yerode how does it compare to other sources in the car? My gain is set on 100% volume, I can make it louder but not without distortion. This was a problem in early v11 builds but they fixed it mostly.

**Reply by yerode** [reply]
@mikegapinski reg: other sources // I tested again with "Tunein Radio" App. It seems that interestingly DVB Radio is much louder than "TuneIn Radio". When using "TuneIn" you can hear Google Maps commands. Okay, of course the Tesla isn't muting the TuneIn sound when a voice command from Google Maps is incoming but it is roughly the same volume/loudness. Have not tested with Apple Music or Spotify. So it's usable with other Tesla sources like TuneIn. Okay. ¯\_(ツ)_/¯ Thanks for your help.

**Reply by mikegapinski** [reply]
@yerode Good news: with the audio logic moved to Dart I will be able to control the gain of the output. You should be able to "boost" the volume of Tesla Android with this to match the radio. The sound of DVB/FM/AM differs depending on the source, Tesla did not implement output leveling like Apple on iOS ;)
